client/src/pages/Campaigns.js

```diff
@@ -48,41 +48,72 @@ const Campaigns = () => {
   const classes = useStyles();
 
   const [open, setOpen] = useState(false); 
-  
+  const [selectedId, setSelectedId] = useState(0);
```

Comment:
Does this ever get used for storing state? Looks like as soon as it gets set you do the redirect.